samza-core/src/main/java/org/apache/samza/processor/SamzaContainerController.java

```diff
@@ -106,7 +109,17 @@ public void startContainer(ContainerModel containerModel, Config config, int max
         Util.<String, MetricsReporter>javaMapAsScalaMap(metricsReporterMap),
         taskFactory);
     log.info("About to start container: " + containerModel.getContainerId());
-    containerFuture = executorService.submit(() -> container.run());
+    containerFuture = executorService.submit(() -> {
+        try {
+          container.run();
+        } catch (Throwable t) {
+          if (errorHandler != null) {
+            errorHandler.onProcessorError(t);
+          } else {
+            throw t;
```

Comment:
Question: If this is running in a different thread, does it still make sense to throw the error? 
Will it make sense to log this error? (Or is logging handled higher or lower up the stack)?